File: CivitasRuby/lib/casilla_calle.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaCalle < Casilla
    
    def initialize(tituloPropiedad)
      super(tituloPropiedad.nombre)   
      @tituloPropiedad = tituloPropiedad
    end
    
    def recibe_jugador(iactual, todos)
      if jugador_correcto(iactual,todos)
        informe(iactual,todos)
        jugador = todos[iactual]
        if @tituloPropiedad.tiene_propietario
          @tituloPropiedad.tramitar_alquiler(jugador)
        else
          jugador.puede_comprar_casilla
        end
      end
    end
    
    def to_s
      "Nombre: #{@nombre} \n"
    end
    
    attr_reader :tituloPropiedad
  end
end

File: CivitasJava/src/civitas/CasillaCalle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author luis
 */
public class CasillaCalle extends Casilla{
    private TituloPropiedad titulo;
    
    CasillaCalle(TituloPropiedad titulo){
        super(titulo.getNombre());
        this.titulo = titulo;
    }
    
    TituloPropiedad getTituloPropiedad(){
        return titulo;
    } 
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(iactual, todos)){
            this.informe(iactual,todos);
            Jugador jugador = todos.get(iactual);
            if(titulo.tienePropietario()){
                titulo.tramitarAlquiler(jugador);
            }else{
                 jugador.puedeComprarCasilla();
            }
            
        } 
    }
    
    @Override
    public String toString(){
        String e = "Nombre: " + this.getNombre() + "\n";
        return e;
    }
}
